### DISCUSSION TITLE
GrandOrgue release for Apple chips

### DISCUSSION THREAD

**Discussion by Gabriele993** [GrandOrgue/grandorgue#1153] [General]
Hello everyone!

I am a very happy user of GrandOrgue software. 
I want to ask if there's a version of GrandOrgue in the works
for Apple chips, since it seems that there's only the Intel version
available in the repository. 
Thanks in advance for your patience.
       Gabriele

**Comment by oleg68** [comment]
Unfortunally we aren't able to build Grandorgue for Apple chips due the lack of its support on GitHub. See https://github.com/actions/runner-images/issues/2187 for details.

But any intel version should work on apple chips with `Rosette`.

The last working version is 3.6.6. 

3.6.7 doesn't work on OSx due the #1135 bug.

3.7.0 is comming soon and it works on OSx again.

**Reply by Gabriele993** [reply]
Hello @oleg68 and sorry for the late response.
Thanks for your information. So it's basically GitHub currently which impedes compiling for Apple chips?
Anyway would it be possible to have GrandOrgue run natively when GitHub releases such feature?
Thank you again.
Gabriele

**Reply by oleg68** [reply]
> Anyway would it be possible to have GrandOrgue run natively when GitHub releases such feature?

Indeed. Yes, it would.

**Comment by rousseldenis** [comment]
@oleg68 I've just seen there is an alternative for CI jobs building for M1 chips

https://github.com/actions/runner-images/issues/2187#issuecomment-1295921395

Could we test if that fits our needs?

**Reply by oleg68** [reply]
> @oleg68 I've just seen there is an alternative for CI jobs building for M1 chips
> 
> https://github.com/actions/runner-images/issues/2187#issuecomment-1295921395
> 
> Could we test if that fits our needs?

 No, we can not, because

> But for Cirrus Runners we only do paid plans.

**Comment by lucpotage** [comment]
Is it the same issue for iOS? Have you planned to release GrandOrgue on App Store, that would be awesome!

**Comment by rousseldenis** [comment]
Maybe good news for this year, github has put runners support in its ROADMAP:

![image](https://user-images.githubusercontent.com/19529533/220551734-d7740aca-b843-428d-ba25-3d51ec87b101.png)

https://github.com/orgs/github/projects/4247/views/1

**Comment by MStraeten** [comment]
if you're brave you might try my local m1 build of todays master [GrandOrgue.3.11.0.zip](https://mega.nz/file/3TpyjKyC#QMen3YPNMADTP3ROM_V0pV5pNX-3k5IqD5oroTHoNfA) maybe you need to allow execution via terminal: " xattr -c <path/to/application.app> " since the app is unsigned.
built with macports - so maybe differences to the official builds; so backup your grandorgue files before using it (at least caches needs to be recreated)

**Comment by willeke1234** [comment]
[The M1 macOS runner is available.](https://github.blog/changelog/2024-01-30-github-actions-introducing-the-new-m1-macos-runner-available-to-open-source/)

**Reply by rousseldenis** [reply]
Great! @oleg68

**Comment by oleg68** [comment]
@willeke1234 Could you test [this build](https://github.com/oleg68/GrandOrgue-official/actions/runs/8189485262/artifacts/1306051691) ?

**Reply by willeke1234** [reply]
The app crashes because of `Code signature invalid`. The build scripts need some tweaks to make it work. I'll try to fix the build locally on my Mac first and while I'm at it, make the app bundle a little less outdated. I'm used to an IDE (add files and hit the Run button) and CMake is a mystery to me. I'm trying to understand what's happening but I don't have the knowledge to change the scripts and I could use some help. Should I ask the questions here or should I start a new thread?

**Reply by oleg68** [reply]
Let's continue talking about build for M1 at https://github.com/GrandOrgue/grandorgue/pull/1831
